### PR TITLE
fix(vlm): advance rope positions across multi-token decode windows

### PR DESCRIPTION
**Pull request by gilbert-barajas**
`VLMModelAdapter._position_ids_from_starts` broadcasts each row's start offset across the whole window, so for `seq_len > 1` every position in the window is rope-rotated at the *first* position instead of advancing.

The consuming attention builds its own positions when none are supplied (`mlx_vlm/models/qwen3_vl_moe/language.py`):

```python
if position_ids is None:
    position_ids = mx.arange(cache.offset, cache.offset + L)
    position_ids = mx.tile(mx.expand_dims(position_ids, 0), (3, 1, 1))
```

so the positions this method passes must advance to match. Prefill (`get_rope_index`) and the MiniMax branch already do — the default mRoPE branch was the only position builder in the stack emitting a constant, and it was never updated when multi-token decode windows were added.

### Why it matters

Multi-token windows arise in speculative-decode verify, where the effect is not just a wrong forward: the mis-rotated keys are **committed into the KV cache on accept**.

Measured on Qwen3.6-35B-A3B, verify attention keys diverged O(1) from an autoregressively-built cache while values matched bit-exactly — the signature of a position error, since rope rotates q and k but not v. With positions advancing, a T-row verify matches T single-row decodes exactly (logits and every layer's cache state) on the prefixes we certified.

### Scope

- **Single-token decode is unchanged by construction.** At `seq_len == 1` the step offsets are `[0]`, so positions reduce to the start offset itself — same shape, same dtype, same values as before.
- Affects the **mRoPE default branch only** (Qwen2/2.5/3-VL-class). MiniMax is a no-op; text-only serving does not use this adapter.
- Two call sites, both decode/verify dispatch; the output is consumed only by the attention rope, so there is no path that was compensating for the old behaviour.

### Tests

Adds a regression test covering multi-token advancement on the default branch — previously only the `seq_len == 1` case and the MiniMax branch (already correct) were pinned. Against the unfixed code it fails with `assert 50 == 51.0`; with the fix the full adapter suite passes (32 tests).

### CODE REVIEW

**Comment by gilbert-barajas** [comment]
Re-verified against `main` @ `41772940` (includes #2356) before asking for a look:

- **Cherry-picks clean** onto current `main` — 2 files, +52/−4, no conflicts.
- **Still needed.** `omlx/models/vlm.py:253` on `main` is unchanged: the MiniMax branch advances positions (`starts[:, None] + steps[None, :]`) while the default mRoPE branch still returns `mx.broadcast_to(starts[None, :, None], (3, batch_size, seq_len))` — a constant across the window. That asymmetry within the same function is the bug.
- **Tests green.** Full `tests/test_vlm_model_adapter.py` on `main` + this patch: 32/32 pass. The added regression test fails without the fix (`assert 50 == 51.0`) and passes with it.
- **Single-token decode is unchanged by construction** — at `seq_len == 1` the step offsets are `[0]`, so positions reduce to the start offset with the same shape and dtype as before.

Worth noting the impact isn't a wrong forward in isolation: in speculative-decode verify the mis-rotated keys are **committed into the KV cache on accept**, so the corruption persists past the window. Measured on Qwen3.6-35B-A3B, verify attention keys diverged O(1) from an autoregressively-built cache while values matched bit-exactly — the signature of a position error, since rope rotates q and k but not v.

CI shows `action_required` (0 runs) — the fork-PR approval gate rather than a failure; happy to rebase if anything above has drifted.

**Comment by jundot** [comment]
Thanks for catching this, I had missed it entirely. The failure mode is quiet enough that normal output still looks fine, so it would have kept sitting there.

- Traced it on Qwen3.5-9B with MTP on: the verify windows arrive as L=2..4 and each window start advances by exactly the window length, so the positions inside the window have to advance too.
- Ran a window vs sequential parity check on two mRoPE models. Without the fix the trailing rows collapse onto the first row's prediction (Qwen2-VL-2B: argmax 15/15/13 where sequential decode gives 362/692/18, max logit delta 16.6; Qwen3.5-9B-oQ4: row 3 argmax 25 vs 2110). With the fix every row matches within quantization noise.
- Regular VLM decode only ever calls this with seq_len 1, so nothing changes on that path.

Really good find. I'll merge this.
